### ISSUE TITLE
Vite dev server crashes when HttpClient receives non-200 code

### ISSUE THREAD

**Issue by zip-fa** [angular/angular-cli#28581]
### Command

serve

### Is this a regression?

- [x] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

v17

### Description

Basically, it's a duplicate of https://github.com/angular/angular-cli/issues/26192

But now it's here again with v19, zoneless and outputMode: 'server'.
Is this behaviour expected and not-an-issue on new setup?

### Minimal Reproduction

ng serve

### Your Environment

HttpErrorResponse

**Comment by alan-agius4** [comment]
This is a known issue with zoneless, and it not limited to dev-servers.

**Comment by zip-fa** [comment]
In prod env, server will crash too? Is there any catch error mechanism that can be applied to not stop dev/prod app, but respond with 500 code?

**Comment by alan-agius4** [comment]
Yes, in the production environment, the server will crash because zone.js is not intercepting and handling unhandled errors and rejections.

Since the error is "unhandled," the request's lifecycle will continue in this scenario. For a potential workaround, please refer to this link: https://github.com/angular/angular-cli/pull/28405

**Comment by alan-agius4** [comment]
Moving to the FW repo as following the convo in https://github.com/angular/angular-cli/pull/28405#discussion_r1791871459 the FW should handle the errors.
